### ISSUE TITLE
A question on systemd IPAccounting=yes & IPAddressDeny=any

### ISSUE THREAD

**Issue by evrim** [NixOS/nixpkgs#151770]
### Describe the bug
Hi,

I was trying to jail some of my services and observed a behaviour of systemd.resource-control.

If I setup IPAccount = yes and IPAddressDeny=any without any IPAddressAllow=, the service is not able to reach any network targets over AF_INET/AF_INET6. That is the expected behaviour I guess of systemd.resource-control.

However, if I exec the following,

```
nsenter -a -t "$(systemctl show --property MainPID --value my-service)" -S some_uid -G some_gid /bin/sh
```

then, when I exec ping 8.8.8.8, I am able to reach the given target. I wonder if anybody else observed similar behaviour or rather, is this the expected behaviour or are there other service configs that I am not aware of to make network contained?

I am on nixos-unstable.

@flokli @andir @aszlig @martinetd 

thnx,
evrim.

**Comment by pennae** [comment]
systemd implement IPAddress{Allow,Deny} with seccomp filters set on the processes it starts, not at the namespace level. entering the namespace won't give you those filters, so you can still ping. if you were to exec a child of the daemon process you'd get the filter and traffic would be blocked.

**Comment by evrim** [comment]
That seems to be the case. I wonder the security implications. Are there any common cases where a RCE might be able to make use of this behaviour? Maybe replacing current binary via execve() or so.

**Comment by andir** [comment]
> That seems to be the case. I wonder the security implications. Are there any common cases where a RCE might be able to make use of this behaviour? Maybe replacing current binary via execve() or so.

From the top of my head: They'd have to be able to create a process that *is not* a child of the process that is being exploited. That would required another process in the namespace to exist.

**Comment by evrim** [comment]
Yes, that's correct. I wonder if zombie's might fail us in this case.

https://coderedirect.com/questions/181762/process-re-parenting-controlling-who-is-the-new-parent
https://github.com/opencontainers/runc/issues/971

The use-case described are as follow. The parent process dies, becomes a zombie and children are owned by init. If I am right, fork()'d children are also applied seccomp by systemd therefore this case is not applicable here, right?

**Comment by flokli** [comment]
This probably isn't NixOS-specific. I'd propose taking that question upstream.

**Comment by martinetd** [comment]
@evrim process becoming a child of pid 1 doesn't magically make seccomp filters disappear, it's still a children of the initially limited process.

That being said I agree with @flokli, that's not a nixos/nixpkgs question -- I'll just close this. Feel free to post a link to a systemd issue here if you create one, or reopen if you disagree
